### ISSUE TITLE
is license header in html deliverable accurate/sufficient?

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/chipper#148]
This header gets inserted into the HTML5 file:

```
Copyright 2002-2013, University of Colorado Boulder
PhET Interactive Simulations
Licensed under GPLv3
http://phet.colorado.edu/en/about/licensing
```

It says "GPLv3", but that is typically only the sim-specific code, and common code is licensed under The MIT License.  Should this header be ammended?

**Comment by samreid** [comment]
It was my understanding that the minified simulation HTML file was to be licensed as GPLv3 or Creative Commons Attribution (see https://phet.colorado.edu/en/about/licensing ).  Yes, it is true that we also license common code libraries such as Scenery as MIT, but when they get bundled into the sim we license *that snapshot* as GPLv3/CCA.

Do we need to update the header to include CCA?  That is a question for @kathy-phet or lawyers, I suspect.  Are we doing the same for HTML5 as we did for Java with respect to https://phet.colorado.edu/en/about/licensing ?

**Comment by pixelzoom** [comment]
Assigning to @ariel-phet to bring this up with @kathy-phet.

**Comment by pixelzoom** [comment]
Still waiting on feedback from @ariel-phet and @kathy-phet on this.  This is a prerequisite to moving forward with "Summer 2015 redeploy".

**Comment by ariel-phet** [comment]
@pixelzoom please bring up at status meeting, I cannot answer this question

**Comment by kathy-phet** [comment]
Hi Chris, 

Please modify this header to read as follows for the Regular PhET sims. And add trademark statement. 

Copyright 2002-2015, Regents of the University of Colorado
PhET Interactive Simulations, University of Colorado Boulder
This file is licensed under Creative Commons Attribution 4.0
For alternate source code licensing, see https://github.com/phetsims
http://phet.colorado.edu/en/licensing

The PhET name and PhET logo are registered trademarks of The Regents of the University of Colorado. Permission is granted to use the PhET name and PhET logo only for attribution purposes. Use of the PhET name and/or PhET logo for promotional, marketing, or advertising purposes requires a separate license agreement from the University of Colorado. Contact phethelp@colorado.edu regarding licensing.


And read as this for Interoperable PhET sims: 

Copyright 2002-2015, Regents of the University of Colorado
PhET Interactive Simulations, University of Colorado Boulder
This Interoperable PhET Simulation file requires a license. USE WITHOUT A LICENSE AGREEMENT IS STRICTLY PROHIBITED. 
Contact phethelp@colorado.edu regarding licensing.
http://phet.colorado.edu/en/licensing

Please double check with Aaron on intended link for licensing at the website: 
http://phet.colorado.edu/en/licensing

**Comment by pixelzoom** [comment]
A different header for interoperable sims? Ugh, that's not currently supported. The header is a static part of the template. @kathy-phet is this a requirement?

We just change the licensing link because it was pointing to the Java/Flash license page. I believe that the correct link is http://phet.colorado.edu/en/html-terms-privacy-and-licensing. @aaronsamuel137  and @kathy-phet please take a look an confirm that this looks correct.

**Comment by aaronsamuel137** [comment]
http://phet.colorado.edu/en/html-terms-privacy-and-licensing is the current link, but it will be changed to http://phet.colorado.edu/en/licensing/html in the near future. The old link will redirect, so it should be fine to leave it as is. I can make a note to myself to update the link in chipper master when I address https://github.com/phetsims/website/issues/80

**Comment by kathy-phet** [comment]
I'm open to suggestions, but separate headers seems to make the most sense to me. These two files are not licensed the same way, and the headers will make that apparent. 

If we have the same header, I do not see how it can include the specific license and would just have to have a link for information. Then people would have to know which they were using - a regular sim or an interoperable sim.

**Comment by pixelzoom** [comment]
@kathy-phet I modified the formatting of your comment https://github.com/phetsims/chipper/issues/148#issuecomment-112579271 because it was difficult to differentiate between your comments and what you wanted in the html file. Please review to make sure I got it correct.

I'm also not clear on a couple of other things:

1. URL for alternate source code licensing

> For alternate source code licensing, see https://github.com/phetsims

Why do you want this URL? I see no "alternate source code licensing" information here, this is the homepage for PhET's github account.

2. You didn't specify the trademark statement for Interoperable PhET sims. Does that mean that you don't want it, or was that an accidental omission?

**Comment by kathy-phet** [comment]
Yes, you have it correct . 

re (1): I listed the main github page because they can find the source code from there. Alternatively we could direct them to Aaron's new link: 
http://phet.colorado.edu/en/licensing/html

re (2): I wasn't going to include the trademark statement in the Interoperable Sim, because the license agreement for the Interoperable Sims will almost certainly include a trademark agreement. I was just trying to emphasize the main point there. In the case of CC-BY, its important that the trademark issue be addressed specifically because they need to attibute PhET but that does not grant permission to use the trademark in other ways.

Kathy

**Comment by samreid** [comment]
Assigned to @samreid to see if anything else needs to be done before Summer 2015 redeploys.  The interoperable text may be moved to a separate, lower-priority issue.

**Comment by samreid** [comment]
I asked @kathy-phet about these phrases at the end of the primary license header:

> For alternate source code licensing, see https://github.com/phetsims
> http://phet.colorado.edu/en/licensing

She said that the first link is about GPL/MIT/etc for the original code.
The second link is more for an overview of the licensing. 

She said I could suggest text revisions if necessary.

**Comment by samreid** [comment]
At the moment I am focusing on the license header for the phet.colorado.edu simulations--we can handle the together.js interoperable simulations license header as a separate, lower priority issue.

I have made the following changes and will commit them shortly for review and discussion:

To be explicit about all of the changes, I will provide the top of the HTML file from http://www.colorado.edu/physics/phet/dev/html/hookes-law/1.0.0-dev.18/hookes-law_en.html
(the previous version)

```html

<!DOCTYPE HTML>
<html>
<head>
  <meta charset="utf-8"/>
  <meta http-equiv="X-UA-Compatible" content="IE=edge"/>
  <meta name="viewport" content="initial-scale=1,user-scalable=no,maximum-scale=1"/>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="phet-sim-level" content="production">
  <title>Hooke's Law 1.0.0-dev.18</title>
  <!--
hookes-law
Copyright 2002-2015, University of Colorado Boulder
PhET Interactive Simulations
Licensed under GPLv3
http://phet.colorado.edu/en/html-terms-privacy-and-licensing
-->
</head>
```

And the top of the HTML file for the new version:

```html
<!DOCTYPE HTML>
<html>
<!--
‪Hooke's Law‬ 1.0.0-dev.18
Copyright 2002-2015, Regents of the University of Colorado
PhET Interactive Simulations, University of Colorado Boulder

This file is licensed under Creative Commons Attribution 4.0
For alternate source code licensing, see https://github.com/phetsims
For licenses for third-party software used by this simulation, see below
For more information, see http://phet.colorado.edu/en/licensing

The PhET name and PhET logo are registered trademarks of The Regents of the
University of Colorado. Permission is granted to use the PhET name and PhET logo
only for attribution purposes. Use of the PhET name and/or PhET logo for promotional
marketing, or advertising purposes requires a separate license agreement from the
University of Colorado. Contact phethelp@colorado.edu regarding licensing.
-->
<head>
  <meta charset="utf-8"/>
  <meta http-equiv="X-UA-Compatible" content="IE=edge"/>
  <meta name="viewport" content="initial-scale=1,user-scalable=no,maximum-scale=1"/>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="phet-sim-level" content="production">
  <title>‪Hooke's Law‬ 1.0.0-dev.18</title>
</head>
```

Notably, I have:
1. Moved the license text to the top of the file, instead of buried under the head/meta/title.  @jonathanolson will this cause any problems?
2. I switched from the Github repository name to the English simulation name and added the version number.
3. I incorporated @kathy-phet changes with the following modifications:
- A line break between the copyright statement and licensing paragraph
- Added the line `For licenses for third-party software used by this simulation, see below`
- Added `For more information, see` to `For more information, see http://phet.colorado.edu/en/licensing`

@kathy-phet can you proofread the above and let me know if my modifications are acceptable?  @jonathanolson can you comment on whether putting this in html instead of head will be acceptable?

I will commit my changes shortly and we can continue from there.

**Comment by samreid** [comment]
Committed changes above, I also tested that hooke's-law builds, lints and runs properly with the modifications.  @jonathanolson or @pixelzoom please let me know if you recommend running any other sanity tests to make sure nothing else was disturbed.  Reassigning to @kathy-phet to review and discuss as described in https://github.com/phetsims/chipper/issues/148#issuecomment-115431097

**Comment by kathy-phet** [comment]
Hi Sam, I reviewed the text. Looks good. Please add a comma after "promotional,". 
Thanks, K

**Comment by samreid** [comment]
I added a comma after "promotional" in the commit above.  Reassigning to @jonathanolson to respond to the two questions I raised for him above.  Here for convenience:

> (1) I moved the license text to the top of the file, instead of buried under the head/meta/title. Will this cause any problems?

> (2) please let me know if you recommend running any other sanity tests to make sure nothing else was disturbed.

**Comment by jonathanolson** [comment]
> 1. Moved the license text to the top of the file, instead of buried under the head/meta/title. @jonathanolson will this cause any problems?

That's generally fine, but I recall there's a small (256-512 bytes?) chunk of the top of the file browsers search for certain meta attributes (I think particularly charset).

Some quick googling brought up https://zoompf.com/blog/2009/12/browser-performance-issues-with-charsets, which notes Firefox checks first 2048 bytes and assumes latin-1 (we use utf-8).

> (2) please let me know if you recommend running any other sanity tests to make sure nothing else was disturbed.

I can't think of anything else.

**Comment by samreid** [comment]
Thanks for pointing out that issue @jonathanolson, I'll move this comment after the head, like so:

```js
<!DOCTYPE HTML>
<html>
<head>
  <meta charset="utf-8"/>
  <meta http-equiv="X-UA-Compatible" content="IE=edge"/>
  <meta name="viewport" content="initial-scale=1,user-scalable=no,maximum-scale=1"/>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="phet-sim-level" content="production">
  <title>‪Hooke's Law‬ 1.0.0-dev.18</title>
</head>
<!--
‪Hooke's Law‬ 1.0.0-dev.18
Copyright 2002-2015, Regents of the University of Colorado
PhET Interactive Simulations, University of Colorado Boulder

This file is licensed under Creative Commons Attribution 4.0
For alternate source code licensing, see https://github.com/phetsims
For licenses for third-party software used by this simulation, see below
For more information, see http://phet.colorado.edu/en/licensing

The PhET name and PhET logo are registered trademarks of The Regents of the
University of Colorado. Permission is granted to use the PhET name and PhET logo
only for attribution purposes. Use of the PhET name and/or PhET logo for promotional,
marketing, or advertising purposes requires a separate license agreement from the
University of Colorado. Contact phethelp@colorado.edu regarding licensing.
-->
<body bgcolor="black">
```

(Keeping this in the head in previous versions was causing odd indenting).

I have tested the above by linting, building and launching hookes-law and everything seemed ok.  Will commit shortly.

**Comment by samreid** [comment]
Committed as described above.   I'll create a separate issue for the together license information, in the together repository, and close this issue.
